packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/WebViewHostApiImpl.java

```diff
@@ -47,11 +125,79 @@ public View getView() {
     public void dispose() {
       destroy();
     }
+
+    @Override
+    public void setWebViewClient(WebViewClient webViewClient) {
+      super.setWebViewClient(webViewClient);
+      currentWebViewClient.set(webViewClient);
+
+      final WebChromeClient webChromeClient = currentWebChromeClient.get();
+      if (webChromeClient instanceof WebChromeClientImpl) {
+        ((WebChromeClientImpl) webChromeClient).setWebViewClient(webViewClient);
+      }
+    }
+
+    @Override
+    public void setDownloadListener(DownloadListener listener) {
+      super.setDownloadListener(listener);
+      currentDownloadListener.set(listener);
+    }
+
+    @Override
+    public void setWebChromeClient(WebChromeClient client) {
+      super.setWebChromeClient(client);
+      currentWebChromeClient.set(client);
+    }
+
+    @SuppressLint("JavascriptInterface")
+    @Override
+    public void addJavascriptInterface(Object object, String name) {
+      super.addJavascriptInterface(object, name);
+      if (object instanceof JavaScriptChannel) {
+        javaScriptInterfaces.put(name, new ReleasableValue<>((JavaScriptChannel) object));
```

Comment:
If this replaces a value, the old values won't get `release()` called until the next GC cycle; is that okay, or does any previous value need to be explicitly found and `release`d here?

Reply:
Good catch. Release actually needs to be called because the Dart Object would still need to be disposed.

Reply:
This is why I think `ReleasableValue` should have a `finalize` that calls `release`.